File: workers/python/kitchen_sink_app.py
from __future__ import annotations

from temporalio.client import Client
from temporalio.worker import Worker

from activities import (
    create_client_activity,
    delay_activity,
    heartbeat_activity,
    noop_activity,
    payload_activity,
    retryable_error_activity,
    timeout_activity,
)
from harness import App, WorkerContext, default_client_factory
from kitchen_sink import KitchenSinkWorkflow, NexusHandlerWorkflow
from nexus_service import KitchenSinkNexusServiceHandler


def app() -> App:
    return App(
        worker=build_worker,
        client_factory=default_client_factory,
    )


def build_worker(client: Client, context: WorkerContext) -> Worker:
    return Worker(
        client,
        task_queue=context.task_queue,
        workflows=[KitchenSinkWorkflow, NexusHandlerWorkflow],
        activities=[
            noop_activity,
            delay_activity,
            payload_activity,
            retryable_error_activity,
            timeout_activity,
            heartbeat_activity,
            create_client_activity(
                client,
                context.err_on_unimplemented,
            ),
        ],
        nexus_service_handlers=[KitchenSinkNexusServiceHandler()],
        **context.worker_kwargs,
    )

File: workers/ruby/kitchen_sink_app.rb
# frozen_string_literal: true

require 'temporalio/worker'
require_relative 'activities'
require_relative 'kitchen_sink'
require_relative 'projects/harness'

module KitchenSinkApp
  module_function

  def app
    Harness::App.new(
      worker: method(:build_worker),
      client_factory: Harness.method(:default_client_factory)
    )
  end

  def build_worker(client, context)
    Temporalio::Worker.new(
      client: client,
      task_queue: context.task_queue,
      workflows: [KitchenSinkWorkflow],
      activities: [
        NoopActivity.new,
        DelayActivity.new,
        PayloadActivity.new,
        RetryableErrorActivity.new,
        TimeoutActivity.new,
        HeartbeatActivity.new,
        ClientActivity.new(client, err_on_unimplemented: context.err_on_unimplemented)
      ],
      logger: context.logger,
      **context.worker_kwargs
    )
  end
end
